include/deal.II/grid/grid_tools.h

```diff
@@ -756,6 +756,86 @@ namespace GridTools
   std::vector<typename MeshType::active_cell_iterator>
   compute_ghost_cell_halo_layer (const MeshType &mesh);
 
+  /**
+   * Extract and return the active cell within a @p layer_thickness
+   * around a subdomain (set of active cells) in the @p mesh.
+   * Here, the "subdomain" consists of exactly all of
+   * those cells for which the @p predicate returns @p true.
+   *
+   * The function first computes interior active, boundary cells of the
+   * subdomain that conform to the predicate.
+   * Using compute_bounding_box(), a bounding box is
+   * computed for this subdomain and extended by @p layer_thickness. All the
+   * active cells that are outside the extended bounding box are ignored.
+   * The cells that are inside the extended bounding box are then checked for
+   * their proximity to the interior subdomain boundary cells. This implies
+   * checking the distance between a pair of arbitrarily oriented cells,
+   * which is not trivial in general. To simplify this, the algorithm checks
+   * the distance between the two smallest spheres that contain each cell.
+   * This will definitely result in slightly more cells being marked but
+   * also greatly simplifies the arithmetic complexity of the algorithm.
+   *
+   * @tparam MeshType A type that satisfies the requirements of the
+   * @ref ConceptMeshType "MeshType concept".
+   * @param mesh A mesh (i.e. objects of type Triangulation, DoFHandler,
+   * or hp::DoFHandler).
+   * @param predicate A function  (or object of a type with an operator())
+   * defining the subdomain around which the halo layer is to be extracted. It
+   * is a function that takes in an active cell and returns a boolean.
+   * @param layer_thickness specifies the distance within
+   * which the function searches for active cells from the predicate domain.
+   * @return A list of active cells sharing at least one common vertex with
+   * the predicated subdomain.
+   *
+   * @image html active_cell_layer_within_distance.png
+   *
+   * See compute_active_cell_halo_layer()
+   *
+   * @author Vishal Boddu, Denis Davydov, 2017
+   */
+  template <class MeshType>
+  std::vector<typename MeshType::active_cell_iterator>
+  compute_active_cell_layer_within_distance
+  (const MeshType                                                                    &mesh,
+   const std_cxx11::function<bool (const typename MeshType::active_cell_iterator &)> &predicate,
+   const double                                                                       layer_thickness);
+
+  /**
+   * Extract and return ghost cells which are within a @p layer_thickness
+   * around all locally owned cells.
```

Comment:
I think it is useful to be explicit and being specific what you mean. It's reasonable to say that all active cells not locally owned in a `p::s::T` are "ghost cells", but it is not something that we widely use in the documentation everywhere. So if that's the definition you want to use for the current function, it's useful to spell that out: "This function is primarily intended for the `p::s::T` class where all cells that are not owned by the current processor can be considered as ghost cells; in particular, they do not only form a single layer of cells around the locally owned ones."